include/stl2/detail/iterator/common_iterator.hpp

```diff
@@ -211,13 +211,14 @@ STL2_OPEN_NAMESPACE {
 			return *__stl2::get_unchecked<I>(v_);
 		}
 		decltype(auto) operator*() const
-		noexcept(noexcept(*std::declval<const I&>())) {
+		noexcept(noexcept(*std::declval<const I&>()))
+		requires detail::Dereferenceable<const I&> {
 			STL2_EXPECT(__stl2::holds_alternative<I>(v_));
 			return *__stl2::get_unchecked<I>(v_);
 		}
 		decltype(auto) operator->() const
 		noexcept(noexcept(__common_iterator::operator_arrow(std::declval<const I&>())))
-		requires Readable<I>() {
+		requires Readable<const I>() {
```

Comment:
I'm a bit concerned that we have too many issue resolutions in flight, and something like this will get lost in the shuffle. Although I suppose that won't happen if we're diligent about out issue tagging.

Maybe we should have open issues for the papers, too, so we don't forget to merge them.